api/v1/helpers/verbs/postUtils.js

```diff
@@ -54,6 +54,10 @@ function makePostPromise(params, props, req) {
   } else {
 
     // cache is off and returnUser is false.
+    if (props.modelName === 'Generator') {
+      return props.model.createWithCollectors(toPost);
```

Comment:
we should decompose the "makePostPromise" function so that we can call the *common* code from *every* model without having a bunch of "if model is Generator/Sample/etc." statements in it, and then put the Generator- and Sample-specific stuff in their own controllers

(that could involve changes to the rest of the controllers and the way they call the common POST functions, so if this should be a separate story, we can leave this here for now and tackle that in another sprint)

Reply:
ok story created.

Reply:
For this PR,  are we leaving it this way or are we moving the createWithCollector to the controller?

Reply:
we are leaving it this way for now. This PR is already getting large. I created a story for the moving of all model-related code in makePostPromise